devicetypes/smartthings/zigbee-motion-sensor-light-light.src/led-cpx-light.groovy

```diff
@@ -0,0 +1,187 @@
+		}
+		
+		valueTile("colorName", "device.colorName", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "colorName", label: '${currentValue}'
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel", "colorTempSliderControl", "colorName"]) //
+	}
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	def event = zigbee.getEvent(description)
+	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	if (event) {
+		if (event.name == "level" && event.value == 0) {
+		} else {
+			if (event.name == "colorTemperature") {
+				setGenericName(event.value)
+			}
+			sendEvent(event)
+		}
+	} else if (zigbeeMap.cluster == "0406" && zigbeeMap.attrId == "0000") {
+		def childDevice = getChildDevices()?.find {
+			it.device.deviceNetworkId == "${device.deviceNetworkId}:1"
+		}
+		zigbeeMap.value = zigbeeMap.value.endsWith("01") ? "active" : "inactive"
+		zigbeeMap.name = "motion"
+		childDevice.sendEvent(zigbeeMap)
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && cluster.clusterId == 0x0006 && cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def ping() {
+	return zigbee.onOffRefresh()
+}
+
+def refresh() {
+	zigbee.configureReporting(0x406, 0x0000, 0x18, 30, 600, null) +
+		zigbee.onOffRefresh() +
+		zigbee.levelRefresh() +
+		zigbee.colorTemperatureRefresh() +
+		zigbee.onOffConfig(0, 300) +
+		zigbee.levelConfig()
+}
+
+def configure() {
+	def cmds = delayBetween([
+		sendEvent(name: "checkInterval", value: 2 * 10 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID]),
+		zigbee.configureReporting(0x406, 0x0000, 0x18, 30, 600, null),
+		zigbee.onOffRefresh(),
+		zigbee.levelRefresh(),
+		zigbee.colorTemperatureRefresh(),
+		zigbee.onOffConfig(0, 300),
+		zigbee.levelConfig()
+	])
+	return cmds
+}
+
+def setColorTemperature(value) {
+	value = value as Integer
+	def tempInMired = Math.round(1000000 / value)
+	def finalHex = zigbee.swapEndianHex(zigbee.convertToHexString(tempInMired, 4))
+	
+	List cmds = []  
+	cmds << zigbee.command(COLOR_CONTROL_CLUSTER, MOVE_TO_COLOR_TEMPERATURE_COMMAND, "$finalHex 0000")
+	cmds << zigbee.readAttribute(COLOR_CONTROL_CLUSTER, ATTRIBUTE_COLOR_TEMPERATURE)
+	cmds
+}
+
+//Naming based on the wiki article here: http://en.wikipedia.org/wiki/Color_temperature
+def setGenericName(value) {
+	if (value != null) {
+		def genericName = "White"
+		if (value < 3300) {
+			genericName = "Soft White"
+		} else if (value < 4150) {
+			genericName = "Moonlight"
+		} else if (value <= 5000) {
+			genericName = "Cool White"
+		} else if (value >= 5000) {
+			genericName = "Daylight"
+		}
+		
+	sendEvent(name: "colorName", value: genericName)
+	}
+}
```

Comment:
This won't have any effect in the SmartThings app. If it's crucial, you'd need to create your own custom capability.